### PR TITLE
Fix inconsistency when querying agents by group

### PR DESCRIPTION
**Pull request by RamosFe**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/18331|

## Description

Fixes inconsistency in `groups/{groupId}/agents` when using the `q` parameter and added test cases for the edge cases.

### CODE REVIEW

File: framework/wazuh/tests/test_agent.py [hunk 1]
```diff
@@ -340,6 +340,36 @@ def test_agent_get_agents_in_group(socket_mock, send_mock, mock_get_groups, mock
...
+    """
+    mock_get_groups.return_value = ['default']
+    # Since the decorator is mocked, pass `group_list` using `call_args` from mock
+    get_agents_in_group(group_list=[group], q=q)
+    kwargs = mock_get_agents.call_args.kwargs
+
+    assert kwargs['q'] == expected_q
+
+
+
```
Comment (GGP1) [inline framework/wazuh/tests/test_agent.py:372]: Remove extra line

File: framework/wazuh/tests/test_agent.py [hunk 2]
```diff
@@ -340,6 +340,36 @@ def test_agent_get_agents_in_group(socket_mock, send_mock, mock_get_groups, mock
...
 
+@pytest.mark.parametrize('group, q, expected_q', [
+    ('default', '(name~wazuh,status~active)', 'group=default;(name~wazuh,status~active)'),
+    ('default', 'name~wazuh,status~active', 'group=default;(name~wazuh,status~active)')
+])
+@patch('wazuh.agent.get_agents')
+@patch('wazuh.agent.get_groups')
+@patch('wazuh.core.wdb.WazuhDBConnection._send', side_effect=send_msg_to_wdb)
+@patch('socket.socket.connect')
+def test_agent_get_agents_in_group_formats_q(socket_mock, send_mock, mock_get_groups, mock_get_agents, group,
```
Comment (GGP1) [inline framework/wazuh/tests/test_agent.py:351]:
```suggestion
def test_agent_get_agents_in_group_q_format(socket_mock, send_mock, mock_get_groups, mock_get_agents, group,
```

File: framework/wazuh/agent.py [hunk 3]
```diff
@@ -396,7 +396,8 @@ def get_agents_in_group(group_list: list, offset: int = 0, limit: int = common.D
     if group_list[0] not in system_groups:
         raise WazuhResourceNotFound(1710)
 
-    q = 'group=' + group_list[0] + (';' + q if q else '')
+    q_group = 'group=' + group_list[0]
+    q = q_group + ';' + (q if q.startswith('(') and q.endswith(')') else f'({q})') if q else q_group
```
Comment (GGP1) [inline framework/wazuh/agent.py:400]: I think it could be easier to read if we do

```suggestion
    if q and not q.startswith('(') and not q.endswith(')'):
        q = f'({q})'
    q = f'group={group_list[0]};' + q
```

**Review by GGP1 — CHANGES_REQUESTED** [review]
We should consider if it's necessary to add two integration test cases to the endpoint `GET /groups/default/agents` with the parameters `q=name~wazuh,status~active` and `q=(name~wazuh,status~active)` to make sure that it returns the results expected.

File: framework/wazuh/tests/test_agent.py (hunk 1 above)
Comment (RamosFe) [reply framework/wazuh/tests/test_agent.py]: Fixed in commit https://github.com/wazuh/wazuh/commit/ca8901ea4735ed4d252571ca378063bb780fdf2.

File: framework/wazuh/tests/test_agent.py (hunk 2 above)
Comment (RamosFe) [reply framework/wazuh/tests/test_agent.py]: https://github.com/wazuh/wazuh/blob/ca8901ea4735ed4d252571ca378063bb780fdf21/framework/wazuh/tests/test_agent.py#L351

File: framework/wazuh/agent.py (hunk 3 above)
Comment (RamosFe) [reply framework/wazuh/agent.py]: Simplified logic:
https://github.com/wazuh/wazuh/blob/ca8901ea4735ed4d252571ca378063bb780fdf21/framework/wazuh/agent.py#L399-L402

**Review by GGP1 — DISMISSED** [review]
LGTM

File: framework/wazuh/agent.py [hunk 4]
```diff
@@ -355,7 +355,10 @@ def get_agents_in_group(group_list, offset=0, limit=common.DATABASE_LIMIT, sort=
     if group_list[0] not in system_groups:
         raise WazuhResourceNotFound(1710)
 
-    q = 'group=' + group_list[0] + (';' + q if q else '')
+    q_group = f'group={group_list[0]}'
+    if q and not (q.startswith('(') and q.endswith(')')):
+        q = f'({q})'
+    q = f'{q_group};{q}' if q else q_group
```
Comment (Selutario) [inline framework/wazuh/agent.py:361]: The bug still exists in queries where multiple parentheses are used:
```
(name~wazuh),(status~active)
``` 

There should be no problem in sending multiple nested parenthesis to `wazuh-db` (`sqlite3`), so this would be better:
```suggestion
    q_group = f'group={group_list[0]}'
    q = f'{q_group};({q})' if q else q_group
```

**Review by Selutario — CHANGES_REQUESTED** [review]

**Review by Selutario — COMMENTED** [review]
Also, please, rebase to 4.7.1 since 4.5.3 status is `testing` and we cannot merge new developments to it.

File: framework/wazuh/agent.py (hunk 4 above)
Comment (RamosFe) [reply framework/wazuh/agent.py]: Changed the logic to this one in commit 6b1246dfe07b048e38acb667376b51c3c23e52ec.

Comment (RamosFe) [reply framework/wazuh/agent.py]:
```
(venv) federamos@pop-os:~/Documents/Wazuh/Repositories/wazuh$ PYTHONPATH=/home/federamos/Documents/Wazuh/Repositories/wazuh/framework/:/home/federamos/Documents/Wazuh/Repositories/wazuh/api pytest framework/wazuh/tests/test_agent.py -vv --disable-warnings
================================================================================= test session starts =================================================================================
platform linux -- Python 3.9.16, pytest-5.4.3, py-1.10.0, pluggy-0.13.1 -- /home/federamos/Documents/Wazuh/Repositories/wazuh/venv/bin/python
cachedir: .pytest_cache
metadata: {'Python': '3.9.16', 'Platform': 'Linux-6.2.6-76060206-generic-x86_64-with-glibc2.35', 'Packages': {'pytest': '5.4.3', 'pluggy': '0.13.1'}, 'Plugins': {'metadata': '2.0.4', 'html': '2.1.1', 'tavern': '1.2.2', 'asyncio': '0.15.1', 'cov': '2.12.0'}}
rootdir: /home/federamos/Documents/Wazuh/Repositories/wazuh/framework
plugins: metadata-2.0.4, html-2.1.1, tavern-1.2.2, asyncio-0.15.1, cov-2.12.0
collected 122 items                                                                                                                                                                   

framework/wazuh/tests/test_agent.py::test_agent_get_distinct_agents[fields0-expected_items0] PASSED                                                                             [  0%]
framework/wazuh/tests/test_agent.py::test_agent_get_distinct_agents[fields1-expected_items1] PASSED                                                                             [  1%]
framework/wazuh/tests/test_agent.py::test_agent_get_distinct_agents[fields2-expected_items2] PASSED                                                                             [  2%]
framework/wazuh/tests/test_agent.py::test_agent_get_distinct_agents[fields3-expected_items3] PASSED                                                                             [  3%]
framework/wazuh/tests/test_agent.py::test_agent_get_distinct_agents[fields4-expected_items4] PASSED                                                                             [  4%]
framework/wazuh/tests/test_agent.py::test_agent_sort_order[fields0-asc-expected_items0] PASSED                                                                                  [  4%]
framework/wazuh/tests/test_agent.py::test_agent_sort_order[fields1-asc-expected_items1] PASSED                                                                                  [  5%]
framework/wazuh/tests/test_agent.py::test_agent_sort_order[fields2-desc-expected_items2] PASSED                                                                                 [  6%]
framework/wazuh/tests/test_agent.py::test_agent_get_agents_summary_status PASSED                                                                                                [  7%]
framework/wazuh/tests/test_agent.py::test_agent_get_agents_summary_os PASSED                                                                                                    [  8%]
framework/wazuh/tests/test_agent.py::test_agent_reconnect_agents[agent_list0-expected_items0-None] PASSED                                                                       [  9%]
framework/wazuh/tests/test_agent.py::test_agent_reconnect_agents[agent_list1-expected_items1-1703] PASSED                                                                       [  9%]
framework/wazuh/tests/test_agent.py::test_agent_reconnect_agents[agent_list2-expected_items2-1701] PASSED                                                                       [ 10%]
framework/wazuh/tests/test_agent.py::test_agent_restart_agents[agent_list0-expected_items0-None] PASSED                                                                         [ 11%]
framework/wazuh/tests/test_agent.py::test_agent_restart_agents[agent_list1-expected_items1-1703] PASSED                                                                         [ 12%]
framework/wazuh/tests/test_agent.py::test_agent_restart_agents[agent_list2-expected_items2-1701] PASSED                                                                         [ 13%]
framework/wazuh/tests/test_agent.py::test_agent_restart_agents_by_node[agent_list0-expected_items0-1703] PASSED                                                                 [ 13%]
framework/wazuh/tests/test_agent.py::test_agent_restart_agents_by_node[agent_list1-expected_items1-1701] PASSED                                                                 [ 14%]
framework/wazuh/tests/test_agent.py::test_agent_get_agents[agent_list0-expected_items0] PASSED                                                                                  [ 15%]
framework/wazuh/tests/test_agent.py::test_agent_get_agents[agent_list1-expected_items1] PASSED                                                                                  [ 16%]
framework/wazuh/tests/test_agent.py::test_agent_get_agents_in_group[default-True-expected_agents0] PASSED                                                                       [ 17%]
framework/wazuh/tests/test_agent.py::test_agent_get_agents_in_group[not_exists_group-False-None] PASSED                                                                         [ 18%]
framework/wazuh/tests/test_agent.py::test_agent_get_agents_in_group_q_formats[default-(name~wazuh,status~active)-group=default;((name~wazuh,status~active))] PASSED             [ 18%]
framework/wazuh/tests/test_agent.py::test_agent_get_agents_in_group_q_formats[default-name~wazuh,status~active-group=default;(name~wazuh,status~active)] PASSED                 [ 19%]
framework/wazuh/tests/test_agent.py::test_agent_get_agents_keys[agent_list0-expected_items0] PASSED                                                                             [ 20%]
framework/wazuh/tests/test_agent.py::test_agent_get_agents_keys[agent_list1-expected_items1] PASSED                                                                             [ 21%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list0-filters0-None-None-expected_items0] PASSED                                                            [ 22%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list1-filters1-None-1731-expected_items1] PASSED                                                            [ 22%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list2-filters2-None-1731-expected_items2] PASSED                                                            [ 23%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list3-filters3-None-1731-expected_items3] PASSED                                                            [ 24%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list4-filters4-None-1731-expected_items4] PASSED                                                            [ 25%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list5-filters5-None-1731-expected_items5] PASSED                                                            [ 26%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list6-filters6-None-1731-expected_items6] PASSED                                                            [ 27%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list7-filters7-None-1731-expected_items7] PASSED                                                            [ 27%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list8-filters8-None-1731-expected_items8] PASSED                                                            [ 28%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list9-filters9-None-1731-expected_items9] PASSED                                                            [ 29%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list10-filters10-None-1731-expected_items10] PASSED                                                         [ 30%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list11-filters11-None-1731-expected_items11] PASSED                                                         [ 31%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list12-filters12-None-1731-expected_items12] PASSED                                                         [ 31%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list13-filters13-None-1731-expected_items13] PASSED                                                         [ 32%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list14-filters14-manager=master;registerIP!=any-1731-expected_items14] PASSED                               [ 33%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list15-filters15-None-1703-expected_items15] PASSED                                                         [ 34%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list16-filters16-None-1701-expected_items16] PASSED                                                         [ 35%]
framework/wazuh/tests/test_agent.py::test_agent_delete_agents[agent_list17-filters17-None-error_code17-None] PASSED                                                             [ 36%]
framework/wazuh/tests/test_agent.py::test_agent_add_agent[agent-1-011-b3650e11eba2f27er4d160c69de533ee7eed601636a85ba2455d53a90927747f-None] PASSED                             [ 36%]
framework/wazuh/tests/test_agent.py::test_agent_add_agent[agent-1-012-b3650e11eba2f27er4d160c69de533ee7eed601636a85ba2455d53a90927747f-force1] PASSED                           [ 37%]
framework/wazuh/tests/test_agent.py::test_agent_add_agent[aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa-002-f304f582f2417a3fddad69d9ae2b4f3b6e6fda788229668af9a6934d454ef44d-None] PASSED [ 38%]
framework/wazuh/tests/test_agent.py::test_agent_get_agent_groups[group_list0-None-expected_result0] PASSED                                                                      [ 39%]
framework/wazuh/tests/test_agent.py::test_agent_get_agent_groups[group_list1-None-expected_result1] PASSED                                                                      [ 40%]
framework/wazuh/tests/test_agent.py::test_agent_get_agent_groups[group_list2-name=group-empty-expected_result2] PASSED                                                          [ 40%]
framework/wazuh/tests/test_agent.py::test_agent_get_agent_groups[group_list3-mergedSum=a336982f3c020cd558a16113f752fd5b-expected_result3] PASSED                                [ 41%]
framework/wazuh/tests/test_agent.py::test_agent_get_agent_groups[group_list4-count=0-expected_result4] PASSED                                                                   [ 42%]
framework/wazuh/tests/test_agent.py::test_agent_get_agent_groups_exceptions[/home/federamos/Documents/Wazuh/Repositories/wazuh/framework/wazuh/tests/data/global.db-invalid_group-1710] PASSED [ 43%]
framework/wazuh/tests/test_agent.py::test_agent_get_group_files[group_list0] PASSED                                                                                             [ 44%]
framework/wazuh/tests/test_agent.py::test_agent_get_group_files[group_list1] PASSED                                                                                             [ 45%]
framework/wazuh/tests/test_agent.py::test_agent_get_group_files_exceptions[/home/federamos/Documents/Wazuh/Repositories/wazuh/framework/wazuh/tests/data/agent/shared-group_list0-False-None-expected_exception0] PASSED [ 45%]
framework/wazuh/tests/test_agent.py::test_agent_get_group_files_exceptions[invalid-path-group_list1-True-None-expected_exception1] PASSED                                       [ 46%]
framework/wazuh/tests/test_agent.py::test_agent_get_group_files_exceptions[/home/federamos/Documents/Wazuh/Repositories/wazuh/framework/wazuh/tests/data/agent/shared-group_list2-True-side_effect2-expected_exception2] PASSED [ 47%]
framework/wazuh/tests/test_agent.py::test_agent_get_group_files_exceptions[/home/federamos/Documents/Wazuh/Repositories/wazuh/framework/wazuh/tests/data/agent/shared-group_list3-True-side_effect3-expected_exception3] PASSED [ 48%]
framework/wazuh/tests/test_agent.py::test_create_group[non-existent-group] PASSED                                                                                               [ 49%]
framework/wazuh/tests/test_agent.py::test_create_group[invalid-group] PASSED                                                                                                    [ 50%]
framework/wazuh/tests/test_agent.py::test_create_group_exceptions[default-WazuhError-1711] PASSED                                                                               [ 50%]
framework/wazuh/tests/test_agent.py::test_create_group_exceptions[group-1-WazuhError-1711] PASSED                                                                               [ 51%]
framework/wazuh/tests/test_agent.py::test_create_group_exceptions[invalid!-WazuhError-1722] PASSED                                                                              [ 52%]
framework/wazuh/tests/test_agent.py::test_create_group_exceptions[delete-me-WazuhInternalError-1005] PASSED                                                                     [ 53%]
framework/wazuh/tests/test_agent.py::test_create_group_exceptions[ar.conf-WazuhError-1713] PASSED                                                                               [ 54%]
framework/wazuh/tests/test_agent.py::test_create_group_exceptions[agent-template.conf-WazuhError-1713] PASSED                                                                   [ 54%]
framework/wazuh/tests/test_agent.py::test_agent_delete_groups[group_list0] PASSED                                                                                               [ 55%]
framework/wazuh/tests/test_agent.py::test_agent_delete_groups[group_list1] PASSED                                                                                               [ 56%]
framework/wazuh/tests/test_agent.py::test_agent_delete_groups_other_exceptions[group_list0-expected_errors0] PASSED                                                             [ 57%]
framework/wazuh/tests/test_agent.py::test_agent_delete_groups_other_exceptions[group_list1-expected_errors1] PASSED                                                             [ 58%]
framework/wazuh/tests/test_agent.py::test_agent_delete_groups_other_exceptions[group_list2-expected_errors2] PASSED                                                             [ 59%]
framework/wazuh/tests/test_agent.py::test_agent_delete_groups_other_exceptions[group_list3-expected_errors3] PASSED                                                             [ 59%]
framework/wazuh/tests/test_agent.py::test_assign_agents_to_group[group_list0-agent_list0-0] PASSED                                                                              [ 60%]
framework/wazuh/tests/test_agent.py::test_assign_agents_to_group[group_list1-agent_list1-1] PASSED                                                                              [ 61%]
framework/wazuh/tests/test_agent.py::test_agent_assign_agents_to_group_exceptions[group_list0-agent_list0-expected_error0-True] PASSED                                          [ 62%]
framework/wazuh/tests/test_agent.py::test_agent_assign_agents_to_group_exceptions[group_list1-agent_list1-expected_error1-False] PASSED                                         [ 63%]
framework/wazuh/tests/test_agent.py::test_agent_assign_agents_to_group_exceptions[group_list2-agent_list2-expected_error2-False] PASSED                                         [ 63%]
framework/wazuh/tests/test_agent.py::test_agent_remove_agent_from_group[default-001] PASSED                                                                                     [ 64%]
framework/wazuh/tests/test_agent.py::test_agent_remove_agent_from_group[group-1-005] PASSED                                                                                     [ 65%]
framework/wazuh/tests/test_agent.py::test_agent_remove_agent_from_group_exceptions[any-group-100-expected_error0] PASSED                                                        [ 66%]
framework/wazuh/tests/test_agent.py::test_agent_remove_agent_from_group_exceptions[any-group-000-expected_error1] PASSED                                                        [ 67%]
framework/wazuh/tests/test_agent.py::test_agent_remove_agent_from_group_exceptions[group-1-005-expected_error2] PASSED                                                          [ 68%]
framework/wazuh/tests/test_agent.py::test_agent_remove_agent_from_groups[group_list0-agent_list0] PASSED                                                                        [ 68%]
framework/wazuh/tests/test_agent.py::test_agent_remove_agent_from_groups_exceptions[group_list0-agent_list0-expected_error0-True] PASSED                                        [ 69%]
framework/wazuh/tests/test_agent.py::test_agent_remove_agent_from_groups_exceptions[group_list1-agent_list1-expected_error1-True] PASSED                                        [ 70%]
framework/wazuh/tests/test_agent.py::test_agent_remove_agent_from_groups_exceptions[group_list2-agent_list2-expected_error2-False] PASSED                                       [ 71%]
framework/wazuh/tests/test_agent.py::test_agent_remove_agents_from_group[group_list0-agent_list0] PASSED                                                                        [ 72%]
framework/wazuh/tests/test_agent.py::test_agent_remove_agents_from_group_exceptions[group_list0-agent_list0-expected_error0-True] PASSED                                        [ 72%]
framework/wazuh/tests/test_agent.py::test_agent_remove_agents_from_group_exceptions[group_list1-agent_list1-expected_error1-False] PASSED                                       [ 73%]
framework/wazuh/tests/test_agent.py::test_agent_remove_agents_from_group_exceptions[group_list2-agent_list2-expected_error2-False] PASSED                                       [ 74%]
framework/wazuh/tests/test_agent.py::test_agent_get_outdated_agents[agent_list0-outdated_agents0] PASSED                                                                        [ 75%]
framework/wazuh/tests/test_agent.py::test_agent_upgrade_agents[agent_set0-expected_errors_and_items0-result_from_socket0-None-False] PASSED                                     [ 76%]
framework/wazuh/tests/test_agent.py::test_agent_upgrade_agents[agent_set1-expected_errors_and_items1-result_from_socket1-filters1-False] PASSED                                 [ 77%]
framework/wazuh/tests/test_agent.py::test_agent_upgrade_agents[agent_set2-expected_errors_and_items2-result_from_socket2-filters2-False] PASSED                                 [ 77%]
framework/wazuh/tests/test_agent.py::test_agent_upgrade_agents[agent_set3-expected_errors_and_items3-result_from_socket3-None-True] PASSED                                      [ 78%]
framework/wazuh/tests/test_agent.py::test_agent_upgrade_agents[agent_set4-expected_errors_and_items4-result_from_socket4-None-True] PASSED                                      [ 79%]
framework/wazuh/tests/test_agent.py::test_agent_get_upgrade_result[agent_set0-expected_errors_and_items0-result_from_socket0-None-False] PASSED                                 [ 80%]
framework/wazuh/tests/test_agent.py::test_agent_get_upgrade_result[agent_set1-expected_errors_and_items1-result_from_socket1-filters1-False] PASSED                             [ 81%]
framework/wazuh/tests/test_agent.py::test_agent_get_upgrade_result[agent_set2-expected_errors_and_items2-result_from_socket2-filters2-False] PASSED                             [ 81%]
framework/wazuh/tests/test_agent.py::test_agent_get_upgrade_result[agent_set3-expected_errors_and_items3-result_from_socket3-None-True] PASSED                                  [ 82%]
framework/wazuh/tests/test_agent.py::test_agent_get_agent_config[agent_list0-logcollector-internal] PASSED                                                                      [ 83%]
framework/wazuh/tests/test_agent.py::test_agent_get_agent_config_exceptions[agent_list0] PASSED                                                                                 [ 84%]
framework/wazuh/tests/test_agent.py::test_agent_get_agents_sync_group[agent_list0] PASSED                                                                                       [ 85%]
framework/wazuh/tests/test_agent.py::test_agent_get_agents_sync_group_exceptions[agent_list0-expected_error0] PASSED                                                            [ 86%]
framework/wazuh/tests/test_agent.py::test_agent_get_agents_sync_group_exceptions[agent_list1-expected_error1] PASSED                                                            [ 86%]
framework/wazuh/tests/test_agent.py::test_agent_get_file_conf[agent.conf-group_list0] PASSED                                                                                    [ 87%]
framework/wazuh/tests/test_agent.py::test_agent_get_agent_conf[group_list0] PASSED                                                                                              [ 88%]
framework/wazuh/tests/test_agent.py::test_agent_upload_group_file[group_list0] PASSED                                                                                           [ 89%]
framework/wazuh/tests/test_agent.py::test_agent_get_full_overview[agent_list0-group_list0-False-001] PASSED                                                                     [ 90%]
framework/wazuh/tests/test_agent.py::test_agent_get_full_overview[agent_list1-group_list1-False-001] PASSED                                                                     [ 90%]
framework/wazuh/tests/test_agent.py::test_agent_get_full_overview[agent_list2-group_list2-False-002] PASSED                                                                     [ 91%]
framework/wazuh/tests/test_agent.py::test_agent_get_full_overview[agent_list3-group_list3-False-002] PASSED                                                                     [ 92%]
framework/wazuh/tests/test_agent.py::test_agent_get_full_overview[agent_list4-group_list4-False-004] PASSED                                                                     [ 93%]
framework/wazuh/tests/test_agent.py::test_agent_get_full_overview[agent_list5-group_list5-True-None] PASSED                                                                     [ 94%]
framework/wazuh/tests/test_agent.py::test_get_agents_big_env[agent_list0-params0-expected_ids0] PASSED                                                                          [ 95%]
framework/wazuh/tests/test_agent.py::test_get_agents_big_env[agent_list1-params1-expected_ids1] PASSED                                                                          [ 95%]
framework/wazuh/tests/test_agent.py::test_get_agents_big_env[agent_list2-params2-expected_ids2] PASSED                                                                          [ 96%]
framework/wazuh/tests/test_agent.py::test_get_agents_big_env[agent_list3-params3-expected_ids3] PASSED                                                                          [ 97%]
framework/wazuh/tests/test_agent.py::test_get_agents_big_env[agent_list4-params4-expected_ids4] PASSED                                                                          [ 98%]
framework/wazuh/tests/test_agent.py::test_get_agents_big_env[agent_list5-params5-expected_ids5] PASSED                                                                          [ 99%]
framework/wazuh/tests/test_agent.py::test_get_agents_big_env[agent_list6-params6-expected_ids6] PASSED                                                                          [100%]

========================================================================== 122 passed, 13 warnings in 3.61s ===========================================================================
```

**Comment by Selutario** [comment]
There are some failing AIT checks, please review them and open a new issue if they are not related to this PR. Moving this issue to on hold.

**Review by Selutario — APPROVED** [review]
